src/main/java/net/snowflake/client/core/SecureStorageAppleManager.java

```diff
@@ -188,7 +188,9 @@ public SecureStorageStatus deleteCredential(String host, String user) {
 
     private static class ResourceHolder {
       private static final SecurityLib INSTANCE =
-          (SecurityLib) Native.loadLibrary("Security", SecurityLib.class);
+          Constants.getOS() == Constants.OS.MAC
+              ? (SecurityLib) Native.loadLibrary("Security", SecurityLib.class)
+              : null;
```

Comment:
can you elaborate how this can improve code coverage?

Reply:
Also in general, setting a variable to `null` is risky. You might get NPE later if you don't check null before reference.

Reply:
Oh actually this helps us to test `getInstance()` function a little bit more, the function right after this `ResourceHolder` class, not itself. The `if (INSTANCE == null)` in `getInstance()` is never covered since we only run JDBC test on Linux and this is for MacOS (another change in this pr is for windows).

And for returning a null here, it's only for testing purpose. Because in real life, this class should never be used when running on OS other than MacOS. `(SecurityLib) Native.loadLibrary("Security", SecurityLib.class);` will break the program if running on other OS. 

In other words, with this change, if the returned `null` causes an NPE in a situation, then, without the change, `(SecurityLib) Native.loadLibrary("Security", SecurityLib.class);` will also break.

But I think it's a good reminder! I can add a debug/error log here for NPE case, though we have very rare chance to see it happened.

Reply:
I'm not convinced that this is a valid change. This makes code cluttered. Can you exclude this file from the code coverage as an exception? Or can you add mac tests to cover this code path?

Reply:
Okay, I think it would be the best if we can add mac tests. And for the same reason, we need Windows tests as well. Only one concern is whether it's worth to do it. Since we know Java is platform independent, and the usage of JNA(JNI) is almost the only one exception, i.e., we are going to setup test environment on Mac and Windows for no one but this feature.

But the above opinion is in the situation that I don't have an accurate estimation of the effort we need to setup Mac/Windows testing env. What do you think @sfc-gh-stakeda , is it gonna take a lot? Or not so much?

Reply:
> Since we know Java is platform independent
But the JDBC driver has the windows and mac specific code, why not having test jobs? Who guarantees those platform specific code work without the tests?

I would suggest the following steps:
- Exclude those two files from the code coverage result as exception, because currently we don't have the mac and windows test jobs.
- Schedule adding the mac and windows tests. Talk to Srinath about the priority. Work with Ankit about how to update the jobs.

At the end of the day, it is up to you and the maintainers. Do you want to maintain such code in the future? Are you ok exposing a risk of NPE, though theoretically not ever happen? In either case, I want to say the above change is not ideal just the code coverage.

Reply:
Got it. Will revert this change and try scheduling adding mac & windows tests first, thanks for the advice!